opm/simulators/flow/BlackoilModelEbosNldd.hpp

```diff
@@ -0,0 +1,801 @@
+                                                          timer.currentStepLength(),
+                                                          iteration,
+                                                          domain,
+                                                          B_avg,
+                                                          residual_norms);
+        report += model_.wellModel().getDomainWellConvergence(domain, B_avg);
+        return report;
+    }
+
+    //! \brief Returns subdomain ordered according to method and ordering measure.
+    std::vector<int> getSubdomainOrder()
+    {
+        const auto& ebosSimulator = model_.ebosSimulator();
+        const auto& solution = ebosSimulator.model().solution(0);
+
+        std::vector<int> domain_order(domains_.size());
+        switch (model_.param().local_solve_approach_) {
+        case DomainSolveApproach::GaussSeidel: {
+            switch (model_.param().local_domain_ordering_) {
+            case DomainOrderingMeasure::AveragePressure: {
+                // Use average pressures to order domains.
+                std::vector<std::pair<double, int>> avgpress_per_domain(domains_.size());
+                for (const auto& domain : domains_) {
+                    double press_sum = 0.0;
+                    for (const int c : domain.cells) {
+                        press_sum += solution[c][Indices::pressureSwitchIdx];
+                    }
+                    const double avgpress = press_sum / domain.cells.size();
+                    avgpress_per_domain[domain.index] = std::make_pair(avgpress, domain.index);
+                }
+                // Lexicographical sort by pressure, then index.
+                std::sort(avgpress_per_domain.begin(), avgpress_per_domain.end());
+                // Reverse since we want high-pressure regions solved first.
+                std::reverse(avgpress_per_domain.begin(), avgpress_per_domain.end());
+                for (size_t ii = 0; ii < domains_.size(); ++ii) {
+                    domain_order[ii] = avgpress_per_domain[ii].second;
+                }
+                break;
+            }
+            case DomainOrderingMeasure::Residual: {
+                // Use maximum residual to order domains.
+                const auto& residual = ebosSimulator.model().linearizer().residual();
+                const int num_vars = residual[0].size();
+                std::vector<std::pair<double, int>> maxres_per_domain(domains_.size());
+                for (const auto& domain : domains_) {
+                    double maxres = 0.0;
+                    for (const int c : domain.cells) {
+                        for (int ii = 0; ii < num_vars; ++ii) {
+                            maxres = std::max(maxres, std::fabs(residual[c][ii]));
+                        }
+                    }
+                    maxres_per_domain[domain.index] = std::make_pair(maxres, domain.index);
```

Comment:
You need `<utility>` for `std::make_pair()`, and for `std::pair<>` for that matter.

Reply:
indeed, added.